meta-mbl-distro/scripts/create-update-payload

```diff
@@ -239,108 +286,131 @@ def get_files_for_boot_part(deploy_dir, tinfoil=None):
-    dst_name.
+    machine = get_bitbake_conf_var("MACHINE", tinfoil)
+    if not machine:
+        bb.fatal("Failed to get value of MACHINE variable")
 
-    :param dst_dir Path: the directory into which files will be copied.
-    """
-    for path in paths:
-        if isinstance(path, tuple):
-            src_path = path[0]
-            dst_name = path[1].name
-        else:
-            src_path = path
-            dst_name = path.name
-        # copy2 preserves file metadata and follows symlinks
-        try:
-            shutil.copy2(str(src_path), str(dst_dir / dst_name))
-        except FileNotFoundError:
-            bb.fatal(
-                "File for the {} update component was not found at path {}. "
-                "Have you built the update components?".format(
-                    dst_name, str(src_path)
-                )
+    rootfs_filename = "{}-{}.tar.xz".format(image_name, machine)
+    try:
+        rootfs_image_path = pathlib.Path(deploy_dir, rootfs_filename).resolve()
+    except FileNotFoundError:
+        bb.fatal(
+            "The rootfs image with filename {} was not found. Have you "
+            "provided the correct image-name? Have you built the rootfs "
+            "image and is it located in {}?".format(
+                rootfs_filename, deploy_dir
             )
+        )
 
+    if not rootfs_image_path.exists():
+        bb.fatal(
+            'Can\'t find rootfs image file "{}"'.format(rootfs_image_path)
+        )
+
+    return ArchivedFileSpec(
+        rootfs_image_path, "{}.tar.xz".format(MBL_ROOTFS_ID)
+    )
 
-def copy_to_staging_dir(filename_var_map, staging_dir, tinfoil=None):
-    """
-    Copy files to a staging dir and rename as is appropriate for inclusion into
-    the payload.
 
-    :param filename_var_map dict: mapping of component names to files.
-    :param staging dir Path: path to the dir where the files will be copied to.
+def stage_multi_file_component(staging_dir, file_name, archived_file_specs):
     """
-    for component_name, paths in filename_var_map.items():
-        if isinstance(paths, list):
-            dst_dir = staging_dir / component_name
-            dst_dir.mkdir(exist_ok=True)
-        else:
-            paths = [(paths, pathlib.Path(component_name))]
-            dst_dir = staging_dir
+    Create a tar.xz file for a multi-file component in the staging dir.
+    """
+    tar_path = staging_dir / file_name
+    with tarfile.open(str(tar_path), mode="w:xz") as tar:
+        for file_spec in archived_file_specs:
+            print_verbose(
+                'Adding "{}" to "{}" as "{}"'.format(
+                    file_spec.path, file_name, file_spec.archived_path
+                )
+            )
+            tar.add(str(file_spec.path), arcname=str(file_spec.archived_path))
+    print_verbose('Adding "{}" to payload'.format(file_name))
 
-        copy_files_to_subdir(paths, dst_dir)
 
+def read_chunks(f):
```

Comment:
This is so that we don't read the whole file into memory at once. I was hoping there'd be a simpler way to do this, but I can't see how I can leverage the buffering in FileIO (or any of the other classes in the io module) to do this.